### PR TITLE
[#2952] feat(core): support name spec capability

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

 - add name spec for entity name

### Why are the changes needed?

Fix: #2952

### Does this PR introduce _any_ user-facing change?

yes, have more limitation in name now

### How was this patch tested?

added tests

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/connector/capability/Capability.java [hunk 1]
```diff
@@ -76,6 +80,22 @@ default CapabilityResult managedStorage(Scope scope) {
...
+     * Regular expression explanation:
+     *
+     * <p>^[a-zA-Z_] - Starts with a letter or an underscore
+     *
+     * <p>[a-zA-Z0-9_/=-]{0,63} - Followed by 0 to 63 characters (making the total length at most
+     * 64) of letters (both cases), digits, underscores, slashes, hyphens, or equals signs
+     *
+     * <p>$ - End of the string
+     */
+    private static final String LEGAL_PATTERN = "^[a-zA-Z_][a-zA-Z0-9_/=-]{0,63}$";
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/connector/capability/Capability.java:97]: Can we rename to DEFAULT_NAME_PATTERN?

File: core/src/main/java/com/datastrato/gravitino/connector/capability/Capability.java [hunk 2]
```diff
@@ -93,6 +113,15 @@ public CapabilityResult caseSensitiveOnName(Scope scope) {
...
     public CapabilityResult specificationOnName(Scope scope, String name) {
+      if (RESERVED_WORDS.contains(name.toLowerCase())) {
+        return CapabilityResult.unsupported(
+            String.format("The %s name '%s' is reserved.", scope, name));
+      }
+
+      if (!name.matches(LEGAL_PATTERN)) {
+        return CapabilityResult.unsupported(
+            String.format("The %s name '%s' is illegal.", scope, name));
+      }
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/connector/capability/Capability.java:124]: Can you please add more tests to test different names.

Comment (mchades) [reply core/src/main/java/com/datastrato/gravitino/connector/capability/Capability.java]: added

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**